src/dense/generic_dense.jl

```diff
@@ -324,6 +340,7 @@ function ode_interpolation(tvals, id::I, idxs, deriv::D, p,
     @unpack ts, timeseries, ks, f, cache = id
     @inbounds tdir = sign(ts[end] - ts[1])
     idx = sortperm(tvals, rev = tdir < 0)
+    differential_vars = get_differential_vars(f, size(timeseries[begin]))
```

Comment:
This should maybe be passed somewhere, but @oscardssmith had some argument why not all cache types need it that I forgot so idk if deep down in the evaluate_interpolant machinery there is somewhere this needs to go

Reply:
we do want to cache this eventually, but at first it's not necessary.